filterflex/client.go

```diff
@@ -70,3 +75,106 @@ func (c *Client) ForwardRequest(req *http.Request) (*http.Response, error) {
 
 	return c.health.Client.Do(req.Context(), proxyReq)
 }
+
+// newRequest creates a new http.Request with auth headers
+func newRequest(ctx context.Context, method string, url string, body io.Reader, userAuthToken, serviceAuthToken, ifMatch string) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create request")
+	}
+
+	if err := headers.SetAuthToken(req, userAuthToken); err != nil {
+		return nil, errors.Wrap(err, "failed to set auth token header")
+	}
+
+	if err := headers.SetServiceAuthToken(req, serviceAuthToken); err != nil {
+		return nil, errors.Wrap(err, "failed to set service token header")
+	}
+
+	if err = headers.SetIfMatch(req, ifMatch); err != nil {
+		return nil, fmt.Errorf("failed to set if match: %w", err)
+	}
+	return req, nil
+}
+
+func (c *Client) createDeleteOptionRequest(ctx context.Context, input GetDeleteOptionInput) (*http.Request, error) {
+	parsedHostURL, err := url.Parse(c.cfg.HostURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse config host url")
+	}
+
+	urlPath := fmt.Sprintf("/filters/%s/dimensions/%s/options/%s", input.FilterID, input.Dimension, input.Option)
+	urlValues := url.Values{}
+
+	parsedHostURL.Path = urlPath
+	parsedHostURL.RawQuery = urlValues.Encode()
+	reqURL := parsedHostURL.String()
+
+	req, err := newRequest(ctx, http.MethodPost, reqURL, nil, input.UserAuthToken, input.ServiceAuthToken, input.IfMatch)
+	if err != nil {
+		return &http.Request{}, dperrors.New(
+			errors.Wrap(err, "failed to create request"),
+			http.StatusBadRequest,
+			log.Data{},
+		)
+	}
+	return req, nil
+}
+
+func (c *Client) DeleteOption(ctx context.Context, input GetDeleteOptionInput) (eTag string, err error) {
+	uri := fmt.Sprintf("%s/filters/%s", c.health.URL, input.FilterID)
+	clientlog.Do(ctx, "retrieving filter", service, uri)
+
+	logData := log.Data{
+		"method":    http.MethodPost,
+		"filter_id": input.FilterID,
+		"dimension": input.Dimension,
+		"option":    input.Option,
+		"ifMatch":   input.IfMatch,
+	}
+
+	req, err := c.createDeleteOptionRequest(ctx, input)
+	if err != nil {
+		return "", dperrors.New(
+			err,
+			dperrors.StatusCode(err),
+			logData,
+		)
+	}
+
+	clientlog.Do(ctx, "deleting options", service, req.URL.String(), logData)
+
+	resp, err := c.health.Client.Do(ctx, req)
+	if err != nil {
+		return "", dperrors.New(
+			errors.Wrap(err, "failed to get response from filter flex API"),
+			http.StatusInternalServerError,
+			logData,
+		)
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
I believe we should be matching against `204` and not `200`
https://github.com/ONSdigital/dp-cantabular-filter-flex-api/blob/develop/api/filter_dimension_options.go#L270

Reply:
will update